packages/shared/src/__tests__/telemetry.test.ts

```diff
@@ -145,29 +146,168 @@ describe('TelemetryCollector', () => {
       publishableKey: TEST_PK,
     });
 
-    collector.record({ event: 'TEST_EVENT', payload: {} });
-    collector.record({ event: 'TEST_EVENT', payload: {} });
+    collector.record({ event: 'TEST_EVENT', payload: { method: 'useFoo' } });
+    collector.record({ event: 'TEST_EVENT', payload: { method: 'useBar' } });
 
     expect(fetchSpy).toHaveBeenCalled();
 
     fetchSpy.mockRestore();
   });
 
-  test('does not send events if the random seed does not exceed the event-specific sampling rate', async () => {
-    const fetchSpy = jest.spyOn(global, 'fetch');
-    const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.1);
+  describe('with server-side sampling', () => {
+    let windowSpy;
 
-    const collector = new TelemetryCollector({
-      publishableKey: TEST_PK,
+    beforeEach(() => {
+      windowSpy = jest.spyOn(window, 'window', 'get');
     });
 
-    collector.record({ event: 'TEST_EVENT', eventSamplingRate: 0.01, payload: {} });
+    afterEach(() => {
+      windowSpy.mockRestore();
+    });
 
-    jest.runAllTimers();
+    test('does not send events if the random seed does not exceed the event-specific sampling rate', async () => {
+      windowSpy.mockImplementation(() => undefined);
 
-    expect(fetchSpy).not.toHaveBeenCalled();
+      const fetchSpy = jest.spyOn(global, 'fetch');
```

Comment:
Nit: You could also move the `fetchSpy` init to a `let` variable and create the mock in `beforeEach` like with `window.get`

Reply:
I've moved some spies to the root `describe` block!